src/main/java/seedu/address/logic/parser/UnmarkCommandParser.java

```diff
@@ -0,0 +1,31 @@
+package seedu.address.logic.parser;
+
+import static seedu.address.logic.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.MarkCommand;
+import seedu.address.logic.commands.UnmarkCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+
+/**
+ * Parses input arguments and creates a new MarkCommand object
+ */
+public class UnmarkCommandParser implements Parser<UnmarkCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the MarkCommand
+     * and returns a MarkCommand object for execution.
+     *
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public UnmarkCommand parse(String args) throws ParseException {
+        try {
+            Index index = ParserUtil.parseIndex(args);
+            return new UnmarkCommand(index);
+        } catch (ParseException pe) {
+            throw new ParseException(
```

Comment:
Should this be replaced with the `InvalidFormatException` Exception?

Reply:
We can use the self-defined exception within the `parser.exceptions` package instead:

```java
package seedu.address.logic.parser.exceptions;

/**
 * Represents an error that occurs when an input command is not properly formatted.
 */
public class InvalidFormatException extends ParseException {
    public InvalidFormatException(String message) {
        super(message);
    }

    public InvalidFormatException(String message, String commandWord, String usageMessage) {
        super(String.format(message, commandWord) + "\nUsage: " + usageMessage);
    }

    public InvalidFormatException(String message, String commandWord, String usageMessage, Throwable cause) {
        super(String.format(message, commandWord) + "\nUsage: " + usageMessage, cause);
    }
}
```